modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/unit/BatchEngineUnitProcessorImpl.java

```diff
@@ -233,6 +248,51 @@ private String _getObjectEntryClassName(
 		return className;
 	}
 
+	private boolean _isProcessed(BatchEngineUnit batchEngineUnit) {
+		Bundle bundle = _getBundle(batchEngineUnit);
+
+		if (bundle == null) {
+			return false;
+		}
+
+		try {
+			BatchEngineUnitConfiguration batchEngineUnitConfiguration =
+				batchEngineUnit.getBatchEngineUnitConfiguration();
+
+			java.io.File batchMarkerFile = bundle.getDataFile(
+				com.liferay.petra.string.StringUtil.merge(
+					Arrays.asList(
+						FileUtil.getShortFileName(
+							batchEngineUnit.getDataFileName()),
+						batchEngineUnitConfiguration.getCompanyId(),
+						"processed"),
+					"."));
+
+			String lastModifiedString = String.valueOf(
+				bundle.getLastModified());
+
+			if ((batchMarkerFile != null) && batchMarkerFile.exists() &&
+				Objects.equals(
+					FileUtil.read(batchMarkerFile), lastModifiedString)) {
+
+				return true;
+			}
+
+			if (!batchMarkerFile.exists()) {
+				batchMarkerFile.createNewFile();
+			}
+
+			FileUtil.write(batchMarkerFile, lastModifiedString, true);
```

Comment:
What if an error happens in between the steps of the execution? Maybe we could split this into two different methods in the BatchEngineUnit, something like:

`public boolean isProcessed()`

`public void markAsProcessed`

Reply:
I think error handling is harder than that. Because the task can fail asynchronously too. This follows the same pattern we had before marking the file as processed before hand. I think improving that would be a separate story.